File: Advanced Programming Methods/Toy Language C#/A2/Model/Data Structures/IStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A2.Model.Data_Structures
{
    //stack generic interface, wrapper
    //in use by the Execution Stack
    public interface IStack<T>
    {
        void push(T el);
        T pop();
        int length();
        Boolean isEmpty();
        string ToString();
    }

}

File: Advanced Programming Methods/Toy Language JavaFX-Procedures/src/Model/DataStructures/IStack.java
package Model.DataStructures;

import java.util.ArrayList;

//stack generic interface, wrapper
//in use by the Execution Stack
public interface IStack<T> {
    void push(T el);
    T pop() throws MyException;
    int length();
    boolean isEmpty();
    String toString();
    ArrayList<T> get();
}
